core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -1440,10 +1446,13 @@ void cancelAndClearAcquisitionLockTimeoutTask() {
         * @return {@code InFlightState} if update succeeds, null otherwise. Returning state
         *         helps update chaining.
         */
-        private InFlightState tryUpdateState(RecordState newState, boolean incrementDeliveryCount) {
+        private InFlightState tryUpdateState(RecordState newState, boolean incrementDeliveryCount, int maxDeliveryCount) {
```

Comment:
It seems to me that the logic of this method is now identical to the majority of startStateTransition. I think startStateTransition should probably call this method, rather than having duplicate logic.

Reply:
+1

Reply:
+1

Reply:
Thanks for the review. Made the changes in the latest commit